src/CalcViewModel/DataLoaders/CurrencyDataLoader.cpp

```diff
@@ -547,9 +548,15 @@ bool CurrencyDataLoader::TryParseStaticData(_In_ String^ rawJson, _Inout_ vector
     }
 
     // TODO - MSFT 8533667: this sort will be replaced by a WinRT call to sort localized strings
-    sort(begin(staticData), end(staticData), [](CurrencyStaticData unit1, CurrencyStaticData unit2)
+
+    auto loc = std::locale("");  //Use the user-preferred locale to sort country names
+    const std::collate<wchar_t>& coll = std::use_facet<std::collate<wchar_t> >(loc);
+    sort(begin(staticData), end(staticData), [&coll](CurrencyStaticData unit1, CurrencyStaticData unit2)
     {
-        return unit1.countryName < unit2.countryName;
+        auto country1 = unit1.countryName.data();
+        auto country2 = unit2.countryName.data();
+        return coll.compare(country1, country1 + unit1.countryName.length(),
```

Comment:
Any reason you can't check the .length() with country1.length()?

Reply:
`wcslen` is O(n) while wstring::length is O(1)